app/src/main/java/com/hedvig/app/feature/profile/ui/payment/PaymentViewModel.kt

```diff
@@ -17,34 +21,36 @@ import kotlinx.coroutines.launch
 abstract class PaymentViewModel(
     hAnalytics: HAnalytics,
 ) : ViewModel() {
-    protected val _paymentData = MutableLiveData<PaymentQuery.Data>()
-    protected val _payinStatusData = MutableLiveData<PayinStatusQuery.Data>()
-    val data = combineTuple(_paymentData, _payinStatusData)
+    protected val _paymentData = MutableStateFlow<PaymentQuery.Data?>(null)
+    protected val _payinStatusData = MutableStateFlow<Pair<PayinStatusQuery.Data?, PaymentType>?>(null)
+    val data: StateFlow<Pair<PaymentQuery.Data?, Pair<PayinStatusQuery.Data?, PaymentType>?>> =
+        combineState(_paymentData, _payinStatusData, viewModelScope)
 
     abstract fun load()
 
     init {
-        hAnalytics.screenViewPayments()
+        hAnalytics.screenView(AppScreen.PAYMENTS)
     }
 }
 
 class PaymentViewModelImpl(
     private val paymentRepository: PaymentRepository,
     private val payinStatusRepository: PayinStatusRepository,
+    private val featureManager: FeatureManager,
     hAnalytics: HAnalytics,
 ) : PaymentViewModel(hAnalytics) {
 
     init {
         viewModelScope.launch {
             paymentRepository
                 .payment()
-                .onEach { _paymentData.postValue(it.data) }
+                .onEach { _paymentData.value = it.data }
                 .catch { e(it) }
                 .launchIn(this)
 
             payinStatusRepository
                 .payinStatusFlow()
-                .onEach { _payinStatusData.postValue(it.data) }
+                .onEach { _payinStatusData.value = Pair(it.data, featureManager.getPaymentType()) }
```

Comment:
This will now ask the featureManager on each new value that `payingStatusFlow` returns, is this intended? Would we maybe want the result of the feature manager function to be called just once and saved maybe even in its own StateFlow or something like that?

Reply:
This is indeed intended. Calls to `FeatureManager` should generally be pretty cheap to make, making this a low-priority concern, and if we introduce a caching layer in the `ViewModel` here, we risk cache staleness on the `ViewModel`-level.